src/Magento/FunctionalTestingFramework/Util/Validation/NameValidationUtil.php

```diff
@@ -50,4 +68,246 @@ public static function validateName($name, $type)
             throw new XmlException($errorMessage);
         }
     }
+
+    /**
+     * Validates data entity names.
+     *
+     * @param string $str
+     * @param string $filename
+     * @throws Exception
+     * @return void
+     */
+    public function validateDataEntityName($str, $filename = null)
+    {
+        if (!ctype_upper($str[0])) {
+            $message = "The data entity name {$str} should be PascalCase with an uppercase first letter.";
+
+            if ($filename !== null) {
+                $message .= " See file {$filename}.";
+            }
+
+            LoggingUtil::getInstance()->getLogger(self::class)->notification(
+                $message,
+                [],
+                false
+            );
+
+            $this->count++;
+        }
+    }
+
+    /**
+     * Validates data entity key names.
+     *
+     * @param string $str
+     * @param string $filename
+     * @throws Exception
+     * @return void
+     */
+    public function validateDataEntityKey($str, $filename = null)
+    {
+        if (!ctype_lower($str[0])) {
+            $message = "The data entity key {$str} should be camelCase with a lowercase first letter.";
+
+            if ($filename !== null) {
+                $message .= " See file {$filename}.";
+            }
+
+            LoggingUtil::getInstance()->getLogger(self::class)->notification(
+                $message,
+                [],
+                false
+            );
+
+            $this->count++;
+        }
+    }
+
+    /**
+     * Validates metadata operation names.
+     *
+     * @param string $str
+     * @param string $filename
+     * @throws Exception
+     * @return void
+     */
+    public function validateMetadataOperationName($str, $filename = null)
+    {
+        if (!ctype_upper($str[0])) {
+            $message = "The metadata operation name {$str} should be PascalCase with an uppercase first letter.";
+
+            if ($filename !== null) {
+                $message .= " See file {$filename}.";
+            }
+
+            LoggingUtil::getInstance()->getLogger(self::class)->notification(
+                $message,
+                [],
+                false
+            );
+
+            $this->count++;
+        }
+    }
+
+    /**
+     * Validates page names.
+     *
+     * @param string $str
+     * @param string $filename
+     * @throws Exception
+     * @return void
+     */
+    public function validatePageName($str, $filename = null)
```

Comment:
can this be repurposed for all prefix, suffix validations?
```
public function validateEntityNameAffixes($str, $type, $filename = null)
{
$isPrefixAdmin = substr_compare($str, self::PREFIX_ADMIN, 0, strlen(self::PREFIX_ADMIN)) === 0;
$isPrefixAdmin = substr_compare($str, self::PREFIX_STOREFRONT, 0, strlen(self::PREFIX_STOREFRONT)) === 0;
$isSuffix = substr_compare($str, $type, 0, strlen($type)) === 0;
..
 $message = "The {$type} name {$str} should follow the pattern {Admin or Storefront}{Description}Page.";
..
}
```

Reply:
fixed